internal/resources/grafana/resource_alerting_notification_policy.go

```diff
@@ -179,42 +184,53 @@ func policySchema(depth uint) *schema.Resource {
 }
 
 func readNotificationPolicy(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	client := OAPIGlobalClient(meta) // TODO: Support org-scoped policies
+	client, orgID, _ := OAPIClientFromExistingOrgResource(meta, data.Id())
 
 	resp, err := client.Provisioning.GetPolicyTree()
-	if err != nil {
-		return diag.FromErr(err)
+	if err, shouldReturn := common.CheckReadError("notification policy", data, err); shouldReturn {
```

Comment:
Unneeded for notification policy, I think. It should never return 404

Reply:
Yes, there will always be a defined receiver.